clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -2957,7 +2957,9 @@ private void prependHeaderAndWriteToStorageEngine(int partition, byte[] keyBytes
      * This value encoding scheme is used in {@link PartitionConsumptionState#maybeUpdateExpectedChecksum(byte[], Put)} to
      * calculate checksum for all the kafka PUT messages seen so far. Any change here needs to be reflected in that function.
      */
-    if (putValue.position() < ValueRecord.SCHEMA_HEADER_LENGTH) {
+    if ((put.putValue.remaining() == 0) && (put.replicationMetadataPayload.remaining() > 0)) {
```

Comment:
Can you please add some javadoc to highlight this logic is used to handle RMD chunks?